packages/prerender/src/errors.tsx

```diff
@@ -23,3 +25,24 @@ export class GqlHandlerImportError {
     this.stack = this.message
   }
 }
+
+interface JSONParseErrorArgs {
+  query: DocumentNode
+  variables?: Record<string, unknown>
+  result: string
+}
+
+export class JSONParseError extends Error {
+  constructor({ query, variables, result }: JSONParseErrorArgs) {
+    const message =
+      'Could not parse the GraphQL response.' +
+      '\n  Request: ' +
+      print(query).split('\n').join('\n  ') +
```

Comment:
Possibly OK to just print the request name here? The query itself can get quite large and may be confusing.

Reply:
I wanted to print the whole request to give as much context as possible without forcing the developer to go hunt for the exact query in their source code. Where the query will be equally large and confusing anyway.

Also give the developer more confidence that that's the exact query that was executed.

Reply:
Sure - don't think there's a "correct" answer here anyway. I just thought it might be confusing to see a large query there.